test/planwise/component/importer_test.clj

```diff
@@ -75,15 +76,59 @@
         (component/stop importer))
 
       ; Create a new importer component that should reload the jobs state from the previous one
-      (let [importer (importer-service system)]
-
-        ; Check that the next task is correctly restored
-        (let [{task-id :task-id} (taskmaster/next-task importer)]
-          (is (= [1 [:import-sites 1]] task-id)))
+      (let [importer (importer-service system)
+            {task-id :task-id} (taskmaster/next-task importer)
+            {next-id :task-id} (taskmaster/next-task importer)]
+          (is (= [1 [:import-sites 1]] task-id))
+          (is (nil? next-id))
 
         (component/stop importer)))))
 
 
+(deftest resume-importer-job-with-multiple-tasks
+  (timbre/with-level :warn
+    (with-system (system)
+
+      ; Create import job for a dataset with a processing-facilities state
+      (let [importer (importer-service system)
+            user-ident {:user-email "john@doe.com", :user-id 1}
+            job (import-job/restore-job {:state :processing-facilities
+                                         :value (merge import-job/default-job-value
+                                                  {:page-count 1, :collection-id 1, :dataset-id 1,
+                                                   :type-field {:code "type", :options {}},
+                                                   :page 1, :user-ident user-ident,
+                                                   :facility-ids [1 2 3 4 5 6],
+                                                   :facility-count 6})})]
+        (swap! (:jobs importer) assoc 1 job)
+
+        ; Fire 4 concurrent process facility tasks
+        (let [tasks (repeatedly 4 #(taskmaster/next-task importer))]
+          (dotimes [n 4]
+            (is (= [1 [:process-facilities [(inc n)]]]
+                   (:task-id (nth tasks n)))))
```

Comment:
True, using `dotimes` feels less _functional-ly_. Fixed in b2d0a82777c91af43543487d023d8a8284cdc533.